### PR TITLE
Box Firmware

### PR DESCRIPTION
**Pull request by buchansm**
This app provides a provisioning script for Raspbian OS - Lite that installs a WIFI server for the Box that an accompanying mobile box setup app can communicate with.

In the future, this API can be extended to boot up `fula`/`libp2p`.

### CODE REVIEW

**Review by gitaaron — COMMENTED** [review]
Great stuff!  Will add a more thorough review in the near future but cursory glance - perhaps it should go under 'apps' instead of 'libraries'?

Libraries are for client side libraries to be included in clients (eg/ react native / web) to help talk with back-ends.

**Comment by buchansm** [comment]
@gitaaron, sounds good 👍. Any recommendations for the name of the directory? I'm currently calling it `box-firmware` but that might be confusing since it'll be next to the `box` app directory.

**Comment by ghorbani-m** [comment]
@buchansm Great job,
Can we convert the app to TypeScript and clarify params' types? 
Now the Box and Libraries are in TypeScript.

File: apps/firmware/src/services/wifi/index.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+/**
+ * Disconnect from wifi.
+ * No questions asked.
+ */
+export const disconnect = async () => {
+  execIgnoreFail(`sudo wpa_cli -i ${config.IFFACE_CLIENT} DISCONNECT`);
+};
+
+/**
+ * Enable eccess point and call "callback" when it is done.
```
Comment (gitaaron) [inline apps/firmware/src/services/wifi/index.js:174]: 'Enable access'

File: apps/firmware/src/services/wifi/index.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+ *
+ * It won't check if each command went well or not.
+ * I really doesn't matter, as in and out states can vary a lot.
```
Comment (gitaaron) [inline apps/firmware/src/services/wifi/index.js:177]: 'It'

Comment (gitaaron) [reply apps/firmware/src/services/wifi/index.js:177]: What do you mean by 'in and out states' ?

**Pull request merged**